### PR TITLE
Check if node_modules is in sync with npm-shrinkwrap.json

### PR DESCRIPTION
**Pull request by IgorMinar**

### CODE REVIEW

**Comment by IgorMinar** [comment]
@caitp can you please review and test on your machine to see if this behaves as expected?

on the first run it's going to complain about node_modules being in unknown state but after the following `npm install` things should work well.

I intentionally didn't force npm reinstall from gulp so that we don't destroy node_modules when someone is just jumping between branches.

**Comment by caitp** [comment]
If there are packages missing from node_modules/ for whatever reason, the check-deps script will be happy, but npm install may fail depending on which packages are missing, which is maybe not good. Also, the use of `preinstall` is strongly discouraged, might want to run the use-case by @isaacs or something.

File: tools/npm/check-node-modules.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+var fs = require('fs');
+var path = require('path');
+
+var NPM_SHRINKWRAP_FILE = 'npm-shrinkwrap.json';
+var NPM_SHRINKWRAP_CACHED_FILE = 'node_modules/npm-shrinkwrap.cached.json';
+var FS_OPTS = {encoding: 'utf-8'};
+var PROJECT_ROOT = path.join(__dirname, '../../');
+
+
+function checkNodeModules(logOutput, purgeIfStale) {
```
Comment (caitp) [inline tools/npm/check-node-modules.js]: maybe using an object for options is a better idea, in case it ever needs to be extended

**Comment by IgorMinar** [comment]
@caitp this is clearly a workaround for npm/npm#8452 so it doesn't try to be perfect or follow best practices.

The case when node_modules could be partially and cached shrinkwrap matches the one in working directory is super rare IMO.

**Comment by btford** [comment]
I just tested this locally and it seems to work. :+1:

**Pull request merged**